### PR TITLE
[HOLD] Globus workflow allows all valid users to proceed

### LINKED ISSUE #3021: Create user folder for Globus workflow on initial save, whether user has logged into Globus before or not
For those who have not logged into a Stanford Globus account before, making this change will eliminate the need for that user to save the item, log into Globus, then return to H2 and notify us that they have completed that step. So, the user journey for this feature would be the same for all users.

We recently learned from the people at Globus (U. Chicago) that it is possible via the API to create the shared folder and later have the correct permissions assigned automatically once the user has logged into Globus. These are the details they provided to us on how to do this:

— Permissions can be set for a user who has not logged into Globus before
— The API call to get the identity UUID, Auth get_identities call is used. The provision=True option ensures that identity is created, as unclaimed. Globus SDK documentation: https://globus-sdk-python.readthedocs.io/en/stable/services/auth.html#globus_sdk.AuthClient.get_identities 

See also #3021, #3022 and #3049.
See designs at https://projects.invisionapp.com/share/YQ132HXFX2GR#/screens/472346509_Globus_Auto_Account.

### PR DESCRIPTION
**Pull request by lwrubel**
# Why was this change made? 🤔
HOLD since it needs to be deployed to qa/stage for @amyehodge's review in addition to code review.  

Resolves #3021 and #3116.

This removes the following states:
* globus_account_setup
* globus_setup_pending
* globus_setup_first_draft
* globus_setup_version_draft

If user has a valid status in Globus, on saving as draft they will see instructions and link to the endpoint:

Screenshot 2023-06-28 at 5 24 37 PM

If user has an invalid status, they will see an error. This can be re-worded. 
Screenshot 2023-06-28 at 5 24 06 PM

# How was this change tested? 🤨
Testing in QA and integration tests. Unit tests updated/added. 

# Does your change introduce accessibility violations? 🩺
No

⚡ ⚠ Please ensure this change does not introduce accessibility violations (at the WCAG A or AA conformance levels); if it does, include a rationale. See the [Infrastructure accessibility guide](https://github.com/sul-dlss/DeveloperPlaybook/blob/main/best-practices/infra-accessibility.md) for more detail. ⚡

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -63,7 +63,8 @@ gem "druid-tools"
 gem "dry-types"
 gem "edtf"
 gem "faraday", "~> 2.0"
-gem "globus_client", "~> 0.8"
+# gem "globus_client", "~> 0.8"
+gem "globus_client", git: "https://github.com/sul-dlss/globus_client", branch: "t3021-add-folder-access"
```
Comment (lwrubel) [inline Gemfile:67]: This will need to be updated before merging, once the globus_client release is cut.

Comment (lwrubel) [reply Gemfile]: Updated now.

**Review by justinlittman — APPROVED** [review]
All those deletions must have been very satisfying ...

**Comment by lwrubel** [comment]
It really was nice to pull out all those steps.

**Pull request merged**